File: c/armstrong-numbers/src/armstrong_numbers.c
#include <stdlib.h>
#include <stdio.h>
#include <stdbool.h>
#include <math.h>
#include "armstrong_numbers.h"

bool is_armstrong_number(unsigned int candidate)
{
    if (candidate < 10)
    {
        return true;
    }
    int num = candidate;
    unsigned int exponent = log(num) / log(10) + 1;

    int digits[exponent];
    unsigned int i = 0;
    while (num > 0)
    {
        digits[i++] = num % 10;
        num /= 10;
    }

    unsigned int digitsum = 0;
    for (unsigned int i = 0; i < exponent; i++)
    {
        digitsum += pow(digits[i], exponent);
    }
    return candidate == digitsum;
}

File: java/armstrong-numbers/ArmstrongNumbers.java
import java.util.Vector;

class ArmstrongNumbers {
    boolean isArmstrongNumber(int numberToCheck) {
        int n = numberToCheck;
        Vector<Integer> digits = new Vector<Integer>();

        while (n > 0) {
            digits.add(n % 10);
            n /= 10;
        }

        int exponent = digits.size();

        int digitsum = 0;
        for (Integer i: digits) {
            digitsum += (int) Math.pow(i, exponent);
        }

        return numberToCheck == digitsum;
    }
}
